caas/kubernetes/provider/credentials.go

```diff
@@ -16,39 +16,67 @@ const (
 	CredAttrPassword              = "password"
 	CredAttrClientCertificateData = "ClientCertificateData"
 	CredAttrClientKeyData         = "ClientKeyData"
+	Token                         = "Token"
 )
 
-type environProviderCredentials struct{}
-
-// CredentialSchemas is part of the environs.ProviderCredentials interface.
-func (environProviderCredentials) CredentialSchemas() map[cloud.AuthType]cloud.CredentialSchema {
-	return map[cloud.AuthType]cloud.CredentialSchema{
-		cloud.UserPassAuthType: {
-			{
-				CredAttrUsername, cloud.CredentialAttr{Description: "The username to authenticate with."},
-			}, {
-				CredAttrPassword, cloud.CredentialAttr{
-					Description: "The password for the specified username.",
-					Hidden:      true,
-				},
+var caasCredentialSchemas = map[cloud.AuthType]cloud.CredentialSchema{
```

Comment:
This is in kubernetes subpackage so call it kubernetesCredentialSchemas or k8sCredentialSchemas